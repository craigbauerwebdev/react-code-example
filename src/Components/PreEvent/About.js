import LinkWrapper from "Components/LinkWrapper/LinkWrapper";
import React from "react";
import aboutStyles from "./scss/about.module.scss";

const About = () => (
  <div className={aboutStyles.aboutContainer}>
    <section className={aboutStyles.aboutContent}>
      <h1>Lorem ipsum dolor sit amet consectetur</h1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
        commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
        velit esse cillum dolore eu fugiat nulla pariatur.
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
      </p>
    </section>
    <section className={aboutStyles.registerTodayHolder}>
      <div className={aboutStyles.registerToday}>
        <h1>Register Today</h1>
        <p>
          Engage in interactive sessions and virtual networking opportunities
          that drive success
        </p>
      </div>
      <div className={aboutStyles.signUp}>
        <LinkWrapper
          to="/"
          page="About"
          componentType="Signup Link"
          trackingUrl="Sign Me Up"
          componentName="sign me up"
        >
          SIGN ME UP
        </LinkWrapper>
      </div>
    </section>
  </div>
);

export default About;
