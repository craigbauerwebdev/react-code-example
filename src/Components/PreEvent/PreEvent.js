import React, { Fragment, Suspense, lazy } from "react";

import BannerWrapper from "Components/Banners/BannerWrapper";
import Loader from "Components/Loader";
import Meta from "../Meta";
import preEventStyles from "./scss/pre-event.module.scss";

const About = lazy(() => import("./About"));
const PreEventSessionsList = lazy(() => import("./PreEventSessionsList"));
const FAQs = lazy(() => import("Components/FAQ/FAQs"));
const SponsorCards = lazy(() => import("./SponsorCards"));

const PreEvent = () => {
  return (
    <Fragment>
      <Meta pageTitle="PreEvent" />
      <BannerWrapper pageName="pre_event" />
      <Suspense
        fallback={
          <div className={preEventStyles.main}>
            <Loader />
          </div>
        }
      >
        <div className={preEventStyles.main}>
          <About />
          <PreEventSessionsList />
          <FAQs preevent pageName="preevent" />
          <SponsorCards />
        </div>
      </Suspense>
    </Fragment>
  );
};

export default PreEvent;
