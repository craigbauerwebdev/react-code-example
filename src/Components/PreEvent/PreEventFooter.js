import FreemanFooter from "Components/Footer/FreemanFooter";
import LinkWrapper from "Components/LinkWrapper/LinkWrapper";
import React from "react";

export const PreEventFooter = (props) => {
  const socialLinkTemplet = () => {
    return props.socialLinks.map((icon) => (
      <LinkWrapper
        to={icon.link}
        className="footer-social gtm-footer-social"
        external={true}
        key={icon.alt}
        page="PreEvent Footer"
        componentType="Social Icon"
        trackingUrl={icon.link}
      >
        <img alt={icon.alt} src={`/images/social-icons/${icon.img}`} />
      </LinkWrapper>
    ));
  };
  return (
    <div className="preevent-footer-wrapper">
      <div className="footer-container flexwrap">
        <img
          id="footer-logo"
          alt="Freeman"
          src="/images/your-event-white.png"
        />
        <div className="footer-social-container flex1">
          {socialLinkTemplet()}
        </div>
      </div>
      <div className="footer-copy-wrapper">
        <section className="preevent-footer-copy">
          <ul>
            <li>Â© Your Event Company 2020</li>
            <li>
              <LinkWrapper
                to="https://www.freeman.com/privacy-policy"
                external={true}
                page="PreEvent Footer"
                componentType="Code of Conduct"
                trackingUrl="https://www.freeman.com/privacy-policy"
                componentName="code of conduct"
              >
                Code of Conduct
              </LinkWrapper>
            </li>
            <li>
              <LinkWrapper
                to="https://www.freeman.com/terms-of-use"
                external={true}
                page="PreEvent Footer"
                componentType="Terms and Conditions"
                trackingUrl="https://www.freeman.com/terms-of-use"
                componentName="Terms and Conditions"
              >
                Terms and Conditions
              </LinkWrapper>
            </li>
          </ul>
        </section>
        <LinkWrapper
          to="https://www.freeman.com/online-event/?utm_source=OEPDEMO-Virtual-Event-Site&utm_medium=footer-logo&utm_campaign=client-website-link"
          external={true}
          page="PreEvent Footer"
          componentType="Home"
          trackingUrl="/"
          componentName="Home"
        >
          <img src="/images/online-event-pro.svg" alt="Online Event Pro" />
        </LinkWrapper>
      </div>
      <FreemanFooter />
    </div>
  );
};
