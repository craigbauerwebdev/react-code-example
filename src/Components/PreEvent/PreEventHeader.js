import LinkWrapper from "Components/LinkWrapper/LinkWrapper";
import OEPAnalytics from "Components/OEPAnalytics";
import React from "react";

export const PreEventHeader = (props) => (
  <div id="header" className="header">
    <div className="headerinteriordiv">
      <LinkWrapper
        to="/"
        className="primarylogo"
        page="PreEvent Header"
        componentType="Home"
        trackingUrl="/"
      >
        <img src="images/your-event.png" alt=" Your Event" className="image" />
      </LinkWrapper>
      <div className="headerinteriordivright">
        <ul className="div-block">
          <li>
            <LinkWrapper
              to="/"
              external={true}
              className="buttonprimary w-button"
              page="PreEvent Header"
              componentType="Register for 2021"
              trackingUrl="/"
              componentName="register now"
            >
              Register Now
            </LinkWrapper>
          </li>
          <li>
            <OEPAnalytics
              page="PreEvent Header"
              componentType="FAQ"
              url="#event-faq"
              componentName="FAQs"
            >
              <a
                href="#event-faq"
                className="buttonprimary w-button faqs-button"
              >
                FAQS
              </a>
            </OEPAnalytics>
          </li>
        </ul>
      </div>
    </div>
  </div>
);
