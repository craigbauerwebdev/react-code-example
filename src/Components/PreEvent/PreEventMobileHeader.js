import LinkWrapper from "Components/LinkWrapper/LinkWrapper";
import OEPAnalytics from "Components/OEPAnalytics";
import React from "react";
export default class PreEventMobileHeader extends React.Component {
  state = {
    openMenu: false,
  };

  toggleMenu = () => {
    this.setState({
      openMenu: !this.state.openMenu,
    });
  };

  render() {
    const { openMenu } = this.state;
    return (
      <div className="header-container-mobile">
        <section>
          <LinkWrapper
            to="/"
            className="primarylogo"
            page="PreEvent Header"
            componentType="Home"
            trackingUrl="/"
            componentName="Home"
          >
            <img
              src="images/your-event.png"
              alt="Your Event"
              className="image"
            />
          </LinkWrapper>
          <OEPAnalytics
            componentType="Button"
            page="Header"
            url="Toggle prevent header"
            componentName="Toggle prevent header"
          >
            <button className="img-button" onClick={this.toggleMenu}>
              <img
                className="hamburger-icon"
                alt="Toggle Primary Menu"
                src="/images/icons/menu.png"
              />
            </button>
          </OEPAnalytics>
        </section>
        {openMenu ? (
          <section>
            <ul className="header-spacer w-inline-block">
              <li>
                <LinkWrapper
                  to="/"
                  className="buttonprimary w-button"
                  page="PreEvent Header"
                  componentType="Register Now"
                  trackingUrl="/"
                  componentName="Register now"
                >
                  REGISTER NOW
                </LinkWrapper>
              </li>
            </ul>
            {!this.props.user && (
              <ul className="header-spacer w-inline-block">
                <li>
                  <LinkWrapper
                    to="/FAQS"
                    className="buttonprimary w-button faqs-button"
                    page="PreEvent Header"
                    componentType="FAQ"
                    trackingUrl="/FAQS"
                    componentName="FAQs"
                  >
                    FAQS
                  </LinkWrapper>
                </li>
              </ul>
            )}
          </section>
        ) : (
          ""
        )}
      </div>
    );
  }
}
