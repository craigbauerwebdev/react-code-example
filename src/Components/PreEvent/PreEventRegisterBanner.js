import LinkWrapper from "Components/LinkWrapper/LinkWrapper";
import React from "react";
import registerBannerStyles from "./scss/pre-event-register-banner.module.scss";

const PreEventRegisterBanner = () => {
  return (
    <div className={registerBannerStyles.registerBanner}>
      <div className={registerBannerStyles.registerBannerContainer}>
        <h1>Your Event</h1>
        <h2>
          September 7 - 9, 2020
          <br />
          Las Vegas, Nevada
        </h2>
        <LinkWrapper
          to="/register"
          rel="noopener"
          page="PreEvent Header"
          componentType="Register Now"
          trackingUrl="/"
          componentName="register"
        >
          Register
        </LinkWrapper>
      </div>
    </div>
  );
};

export default PreEventRegisterBanner;
