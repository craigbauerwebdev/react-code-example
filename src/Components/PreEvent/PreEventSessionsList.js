import React, { useEffect, useReducer } from "react";
import {
  paginationReducer,
  paginationState,
} from "Components/Paginate/reducer";
import sortResults, { sortTypes } from "util/sortResults";
import { useDispatch, useSelector } from "react-redux";

import Loader from "Components/Loader";
import { Paginate } from "../Paginate/Paginate";
import PreEventSessionsTab from "./PreEventSessionsTab";
import { dataTypes } from "store/utils/dataTypes";
import { getPayload } from "store/actions";
import preEventSessions from "./scss/pre-event-sessions-list.module.scss";

const SessionsList = () => {
  const dispatch = useDispatch();
  /**@type {Session[]} */
  const sessions = useSelector((state) => state.global.sessions);
  const [statePaginationState, dispatchPagination] = useReducer(
    paginationReducer,
    paginationState
  );
  const listRef = React.createRef();

  useEffect(() => {
    window.scrollTo(0, 0);
    if (!sessions) {
      dispatch(getPayload(dataTypes.sessions));
    }
  }, [sessions, dispatch]);

  if (!sessions) {
    return <Loader />;
  }

  return (
    <section>
      <h1 className={preEventSessions.title}>Sessions</h1>

      <div
        className={preEventSessions.results}
        autoFocus={true}
        ref={listRef}
        tabIndex="-1"
      >
        {sortResults(sessions, sortTypes.time)
          .slice(statePaginationState.startIndex, statePaginationState.endIndex)
          .map((data, i) => (
            <PreEventSessionsTab data={data} key={i} index={i} />
          ))}
      </div>

      <Paginate
        large
        inc={7}
        total={sessions.length}
        startIndex={statePaginationState.startIndex}
        dispatch={dispatchPagination}
        listRef={listRef}
        pageTile="Sessions"
        page="preevent"
      />
    </section>
  );
};

export default SessionsList;
