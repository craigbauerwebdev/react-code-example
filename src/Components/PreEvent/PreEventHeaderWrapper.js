import React from "react";
import { PreEventHeader } from "./PreEventHeader";
import PreEventMobileHeader from "./PreEventMobileHeader";

export default class PreEventHeaderWrapper extends React.Component {
  state = {
    tab1: false,
    tab2: false,
    tab3: false,
    tab4: false,
    tab5: false,
  };

  // Header is triggered by hover
  // Mobile is triggered by click

  // To change Header to be triggered by click, remove
  // `.header-label:hover + .inactive-dropdown` and uncomment out class logic on the inactive-dropdown element

  updateOpenMenu = (menu) => {
    const newState = {};
    const setTofalse = Object.keys(this.state).filter((key) => key !== menu);
    const setToTrue = Object.keys(this.state).find((key) => key === menu);

    setTofalse.map((key) => (newState[key] = false));
    newState[setToTrue] = !this.state[setToTrue];
    this.setState(newState);
  };

  render() {
    return (
      <div>
        <PreEventHeader
          user={this.props.user}
          updateOpenMenu={this.updateOpenMenu}
          state={this.state}
        />
        <PreEventMobileHeader
          user={this.props.user}
          updateOpenMenu={this.updateOpenMenu}
          state={this.state}
        />
      </div>
    );
  }
}
