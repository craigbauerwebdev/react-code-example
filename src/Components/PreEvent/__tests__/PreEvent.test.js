import PreEvent from "../PreEvent";
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

jest.mock("Components/Banners/BannerWrapper", () => "BannerWrapper");

jest.mock("Components/Loader", () => "Loader");

jest.mock("Components/Meta", () => "Meta");

jest.mock("Components/PreEvent/About", () => "About");

jest.mock(
  "Components/PreEvent/PreEventSessionsList",
  () => "PreEventSessionsList"
);

jest.mock("Components/FAQ/FAQs", () => "FAQs");

jest.mock("Components/PreEvent/SponsorCards", () => "SponsorCards");

it("renders without crashing", () => {
  const tree = shallow(<PreEvent />);
  expect(tree.find("lazy").get(2).props.pageName).toBe("preevent");
  expect(toJson(tree)).toMatchSnapshot();
});
