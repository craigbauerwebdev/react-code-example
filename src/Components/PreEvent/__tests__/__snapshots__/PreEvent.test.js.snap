// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <Meta
    pageTitle="PreEvent"
  />
  <BannerWrapper
    pageName="pre_event"
  />
  <Suspense
    fallback={
      <div
        className="main"
      >
        <Loader />
      </div>
    }
  >
    <div
      className="main"
    >
      <lazy />
      <lazy />
      <lazy
        pageName="preevent"
        preevent={true}
      />
      <lazy />
    </div>
  </Suspense>
</Fragment>
`;
